src/global/project/mod.rs

```diff
@@ -1050,9 +1055,12 @@ impl Project {
 
 impl Repodata for Project {
     /// Returns the [`Gateway`] used by this project.
-    fn repodata_gateway(&self) -> &Gateway {
-        self.repodata_gateway
-            .get_or_init(|| self.config().gateway(self.authenticated_client().clone()))
+    fn repodata_gateway(&self) -> miette::Result<&Gateway> {
+        // todo: fix unwrap
+        Ok(self.repodata_gateway.get_or_init(|| {
+            self.config()
+                .gateway(self.authenticated_client().unwrap().clone())
```

Comment:
Same here

Reply:
this requires [get_or_try_init](https://doc.rust-lang.org/beta/std/sync/struct.OnceLock.html#method.get_or_try_init), which is currently unstable: https://github.com/rust-lang/rust/issues/109737

see [`899bd0f` (#2825)](https://github.com/prefix-dev/pixi/pull/2825/commits/899bd0f9a3f4ad58db5e0b13e9e29e9979c1e8e3). 
should we still incorporate this change?